README.md

```diff
@@ -48,6 +48,78 @@ $ npm run start:dot
 
 **Detailed information in the [DEVELOPMENT_NOTES.md](/docs/DEVELOPMENT_NOTES.md) document.**
 
+### Debugging contracts
+
+If a call to a contract from the frontend fails, you may not get much information about the failure in the console. One of the best ways to troubleshoot is to tweak the contract code and try again, but restarting the whole development stack is time-consuming. Here's a way to make it faster:
+
+1. Record the info that was sent to the contract from the frontend (you can log the values to your browser console). Example:
+
+   ```js
+   addBounties(["0x4d4445774f6c4a6c6347397a61585276636e6b784d6a59344f546b784e444d3d"], [1234], ["1000000000000000000"], [1568813389132], [0], ["0x0000000000000000000000000000000000000000"], 'Qmb1wpM8Yzs3pDmynGKgMGHkVUcdfnTjWnvTA7LJTuA167', 'Bounty description')
+   ```
+
+2. In your terminal, `cd` into the app folder for the app you're troubleshooting. Example: `cd apps/projects`
+
+3. In that directory, run `truffle console`. This starts a node console with some extra goodies like [web3.js](https://web3js.readthedocs.io), thanks to [truffle](https://www.trufflesuite.com/docs/truffle/getting-started/using-truffle-develop-and-the-console). You can find out what versions of things you're using by typing `version`, so you're sure you're looking at the correct docs.
+
+4. To read and write to your problematic contract, you will need its abi and its address.
+
+   * **Address**: visit your app in your browser (e.g. click on "Projects" in the side bar) and look at the URL. It'll be something like:
+
+     ```
+     http://localhost:3000/#/0xb84dFbdc18069a83af4D5506096f5e7AC7554183/0x4eda83d4a45d4a00bd128a4002e7699490a87649
+     ```
+
+     The last bit there is the address of your smart contract. Save it in a variable in your truffle console:
+
+     ```js
+     addr = '0x4eda83d4a45d4a00bd128a4002e7699490a87649'
+     ```
+
+   * **abi**: When your contract is compiled, it creates an _application binary interface_, which web3.js needs in order to interact with the contract. This is saved in a json file in your app's directory. For the `Projects.sol` contract, the relevant JSON file is at `apps/projects/build/contracts/Projects.json`. To load the abi into your truffle console, you can do something like this:
+
+     ```js
+     let abi; fs.readFile('./build/contracts/Projects.json', 'utf-8', (err, data) => { if (err) { throw err } abi = JSON.parse(data).abi })
+     ```
+
+5. Instantiate your contract:
+
+   ```js
+   contract = new web3.eth.Contract(abi, addr)
+   ```
+
+6. Now you can directly call the method that you saved in step 1:
+
+   ```js
+   contract.methods.addBounties(...).call()
+   ```
+
+   Or query all its events:
+
+   ```js
+   contract.getPastEvents('allEvents')
+   ```
+
+   Or anything else listed in [the web3 docs](https://web3js.readthedocs.io/en/v1.2.1/web3-eth-contract.html)
+
+7. Make changes to your contract code
+
+8. Re-compile & deploy your contract
+
+   ```
+   ???
```

Comment:
@topocount what commands do we want here?

Reply:
We should just remove 7 and 8 steps. Or replace by something like 
> ask a solidity developer to look into it if you think there is an issue with the contract